Moving docs publish ci job

BobaFetters opened the pull request:
Moving the docs publish ci job from the `apollo-ios` repo to the `apollo-ios-dev` repo

calvincestari reviewed (commented):
Do we need to remove there [branches from the `apollo-ios` repo too](https://github.com/apollographql/apollo-ios/blob/main/.github/workflows/docs-publish.yml#L5) then if we're moving publishing to this repo?

BobaFetters commented:
So I removed the `release/1.0` branch from the script when i moved it to this repo, but the main branch should be our target to publish docs from when changes are detected like we were in the other repo.
